java/repositories.bzl

```diff
@@ -502,19 +369,51 @@ def bazel_skylib():
         sha256 = "1dde365491125a3db70731e25658dfdd3bc5dbdfd11b840b3e987ecf043c7ca0",
     )
 
+def rules_python():
+    maybe(
+        http_archive,
+        name = "rules_python",
+        type = "tar.gz",
+        url = "https://github.com/bazelbuild/rules_python/releases/download/0.4.0/rules_python-0.4.0.tar.gz",
+        sha256 = "954aa89b491be4a083304a2cb838019c8b8c3720a7abb9c4cb81ac7a24230cea",
+    )
+
 def rules_java_dependencies():
     """An utility method to load all dependencies of rules_java.
 
     Loads the remote repositories used by default in Bazel.
     """
 
+    local_jdk_repo()
     remote_jdk11_repos()
-    java_tools_javac11_repos()
+    remote_jdk15_repos()
+    remote_jdk16_repos()
+    remote_jdk17_repos()
+
+    # TODO: load this will break compatibility with Bazel 4.2.1
```

Comment:
Why is that? Bazel 4.2.1 is already using 11.x version of java_tools. It should also work with v11.5.

Reply:
The error I'm getting is
```
ERROR: /private/var/tmp/_bazel_pcloudy/f0729e7dd3769854a3aed1679d51c520/external/bazel_tools/tools/jdk/BUILD:378:6: no such target '@remote_java_tools_darwin//:toolchain': target 'toolchain' not declared in package '' defined by /private/var/tmp/_bazel_pcloudy/f0729e7dd3769854a3aed1679d51c520/external/remote_java_tools_darwin/BUILD and referenced by '@bazel_tools//tools/jdk:remote_toolchain'
```

Reply:
In Bazel 4.2.1, @bazel_tools//tools/jdk:remote_toolchain is
```
alias(
    name = "remote_toolchain",
    actual = select({
        "//src/conditions:darwin": "@remote_java_tools_darwin//:toolchain",
        "//src/conditions:windows": "@remote_java_tools_windows//:toolchain",
        "//src/conditions:linux_x86_64": "@remote_java_tools_linux//:toolchain",
        "//conditions:default": "@bazel_tools//tools/jdk:legacy_toolchain",
    }),
)
```
But in Bazel@HEAD, it's
```
default_java_toolchain(
    name = "toolchain",
    configuration = DEFAULT_TOOLCHAIN_CONFIGURATION,
    toolchain_definition = False,
)

alias(
    name = "remote_toolchain",
    actual = ":toolchain",
)
```

Reply:
Ups. Sorry. I made a mistake. Bazel 4.2.1 is using 10.x version of java_tools. There are a lot of other incompatibilities in this change for Bazel 4.2.1. We already tagged rules_java. I wouldn't worry about breaking it at all.

Reply:
OK, no problem. But to not break the CI (which currently use 4.2.1), I'll keep it comment out for now, we can enable it after Bazel 5.0 is released.